core/serialize_deserialize.rs

```diff
@@ -0,0 +1,94 @@
+// Copyright 2018-2021 the Deno authors. All rights reserved. MIT license.
+
+use rusty_v8 as v8;
+
+type ArrayBuffers = Vec<v8::SharedRef<v8::BackingStore>>;
+
+struct SerializeDeserialize<'a> {
+  array_buffers: &'a mut ArrayBuffers,
+}
+
+impl<'a> SerializeDeserialize<'a> {
+  fn serializer<'s>(
+    scope: &mut v8::HandleScope<'s>,
+    array_buffers: &'a mut ArrayBuffers,
+  ) -> v8::ValueSerializer<'a, 's> {
+    v8::ValueSerializer::new(scope, Box::new(Self { array_buffers }))
+  }
+
+  fn deserializer<'s>(
+    scope: &mut v8::HandleScope<'s>,
+    data: &[u8],
+    array_buffers: &'a mut ArrayBuffers,
+  ) -> v8::ValueDeserializer<'a, 's> {
+    v8::ValueDeserializer::new(scope, Box::new(Self { array_buffers }), data)
+  }
+}
```

Comment:
Seems like this struct is superfluous, why not use `v8::ValueSerializer` directly?

Reply:
I just pushed a simplified version.